File: frontend/src/types/prayer.ts
export interface HebrewWord {
  word: string;
  transliteration: string;
  translation: string;
  root?: string;
}

export interface PrayerSection {
  ref: string;
  title: string;
  hebrew: string;
  english: string;
  words: HebrewWord[];
}

export interface ServiceSectionMeta {
  ref: string;
  title: string;
  is_rubric?: boolean;
  group?: string;
}

export interface ServiceInfo {
  id: string;
  name: string;
  hebrew_name: string;
  description: string;
  sections: ServiceSectionMeta[];
}

export interface WordDefinition {
  word: string;
  transliteration: string;
  definitions: string[];
}

File: backend/app/models/prayer.py
from pydantic import BaseModel
from typing import Optional


class HebrewWord(BaseModel):
    word: str
    transliteration: str
    translation: str
    root: Optional[str] = None


class PrayerSection(BaseModel):
    ref: str
    title: str
    hebrew: str
    english: str
    words: list[HebrewWord]


class ServiceInfo(BaseModel):
    id: str
    name: str
    hebrew_name: str
    description: str
    sections: list[dict]
